tools/c7n_azure/tests_azure/test_provider.py

```diff
@@ -27,3 +30,74 @@ def test_initialize_custom_account_id(self):
 
         session = azure.get_session_factory(options)()
         self.assertEqual(sample_account_id, session.get_subscription_id())
+
+    def test_initialize_default_azure_cloud(self):
+        with patch('c7n_azure.session.Session.get_subscription_id'):
+            options = Config.empty()
+            azure = Azure()
+            azure.initialize(options)
+            self.assertEqual(AZURE_PUBLIC_CLOUD, azure.cloud_endpoints)
+            self.assertEqual(AZURE_PUBLIC_CLOUD.name, options['region'])
+            session = azure.get_session_factory(options)()
+
+        self.assertEqual(AZURE_PUBLIC_CLOUD.endpoints.active_directory_resource_id,
+                         session.resource_namespace)
+
+    def test_initialize_default_azure_cloud_flag(self):
+        with patch('c7n_azure.session.Session.get_subscription_id'):
+            options = Config.empty(regions=[DEFAULT_AZURE_CLOUD])
+            azure = Azure()
+            azure.initialize(options)
+            self.assertEqual(AZURE_PUBLIC_CLOUD, azure.cloud_endpoints)
+            self.assertEqual(AZURE_PUBLIC_CLOUD.name, options['region'])
+            session = azure.get_session_factory(options)()
+
+        self.assertEqual(AZURE_PUBLIC_CLOUD.endpoints.active_directory_resource_id,
+                         session.resource_namespace)
+
+    def test_initialize_china_azure_cloud(self):

```

Comment:
You can use `@pytest.mark.parametrize` to pass a set of parameters and have single test